features/FEATURE_LWIP/lwip-interface/EthernetInterface.h

```diff
@@ -100,6 +100,18 @@ class EthernetInterface : public EthInterface
      */
     virtual const char *get_gateway();
 
+    /** Register callback for status reporting
+     *
+     *  @param status_cb The callback for status changes
+     */
+    virtual void attach(mbed::Callback<void(nsapi_connection_status_t, int)> status_cb);
```

Comment:
The callback here isn't following the discussion we had in that 2017-11-08 meeting - that agreed generalising to be effectively `(nsapi_event_t reason, uint32_t parameter)`, so for this initial event the "reason" would be "connection status change", and the parameter would be the `nsapi_connection_status_t`.